src/item_proxy.rs

```diff
@@ -0,0 +1,62 @@
+//Copyright 2020 secret-service-rs Developers
+//
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+//! A dbus proxy for speaking with secret service's `Item` Interface.
+
+use serde::{Deserialize, Serialize};
+use zbus;
+use zbus_macros::dbus_proxy;
+use zvariant::{Dict, ObjectPath, OwnedObjectPath};
+use zvariant_derive::Type;
+
+/// This will derive ItemInterfaceProxy
+#[dbus_proxy(
+    interface = "org.freedesktop.Secret.Item",
+)]
+trait ItemInterface {
+    fn delete(&self) -> zbus::Result<OwnedObjectPath>;
+
+    /// returns `Secret`
+    fn get_secret(&self, session: ObjectPath) -> zbus::Result<SecretStruct>;
+
+    fn set_secret(&self, secret: SecretStructInput) -> zbus::Result<()>;
```

Comment:
Any reason for not using SecretStruct directly here?

Reply:
I'm not quite sure what's going on in secret service; I thought that just passing the struct as `(oayays)` was fine, but I was getting an error that the method expected `((oayays))`. After wrapping, the error went away, but this is something I should revisit. I don't think I had to do this workaround in my previous code.

Reply:
This might be useful https://gitlab.freedesktop.org/zeenix/zbus/-/issues/104

Reply:
hmm.. did you tested with zbus >= `1.3.0`? I was hoping this issue was behind us.

Reply:
I did. I'll look into that issue, and also double check the error message.

Reply:
very strange. We even have a [testcase for this](https://gitlab.freedesktop.org/zeenix/zbus/-/blob/master/zbus/src/object_server.rs#L658).